os/android/libhwcservice/utils/mode.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include <binder/IServiceManager.h>
+#include <utils/String8.h>
+#include <cinttypes>
+#include "hwcserviceapi.h"
+#include "iservice.h"
+
+using namespace android;
+using namespace hwcomposer;
+
+int main(int argc, char** argv) {
+  // Argument parameters
+  uint32_t display = 0;
+  uint32_t displayModeIndex = 0;
+  bool printModes = false;
+  bool bPreferred = false;
+  bool getMode = false;
+  bool setMode = false;
+  int argIndex = 1;
+  int nonOptions = 0;
+
+  // Process options
+  while (argIndex < argc) {
+    // Process any non options
+    switch (nonOptions) {
+      case 0:
+        display = atoi(argv[argIndex]);
+        break;
+      case 1:
+        printModes = false;
+        if (strcmp(argv[argIndex], "get") == 0) {
+          getMode = true;
+        } else if (strcmp(argv[argIndex], "set") == 0) {
+          setMode = true;
+          displayModeIndex = atoi(argv[argIndex++]);
+        } else if (strcmp(argv[argIndex], "print") == 0)
+          printModes = true;
+        break;
+    }
+    nonOptions++;
+    argIndex++;
+  }
+
+  if (nonOptions == 0) {
+    printf("Usage: %s  [displayId <print/get/set <displayconfigindex>>]\n",
+           argv[0]);
+    return 1;
+  }
+
+  // Find and connect to HWC service
+  sp<IService> hwcService = interface_cast<IService>(
```

Comment:
hwc service is based on Android binders which are strong pointers. I will get compilation errors, If I change it to other pointers.